app/scripts/services/data-deletion-service.ts

```diff
@@ -0,0 +1,274 @@
+    analyticsDataDeletionEndpoint?: string;
+    analyticsDataDeletionSourceId?: string;
+    circuitBreakDuration?: number;
+    degradedThreshold?: number;
+    onBreak?: () => void;
+    onDegraded?: () => void;
+    timeout?: number;
+  } = {}) {
+    if (!analyticsDataDeletionEndpoint) {
+      throw new Error('Missing ANALYTICS_DATA_DELETION_ENDPOINT');
+    } else if (!analyticsDataDeletionSourceId) {
+      throw new Error('Missing ANALYTICS_DATA_DELETION_SOURCE_ID');
+    }
+    this.#fetchWithTimeout = getFetchWithTimeout(timeout);
+    this.#analyticsDataDeletionEndpoint = analyticsDataDeletionEndpoint;
+    this.#analyticsDataDeletionSourceId = analyticsDataDeletionSourceId;
+    this.#createDataDeletionTaskPolicy = createRetryPolicy({
+      circuitBreakDuration,
+      degradedThreshold,
+      maximumConsecutiveFailures: MAX_CONSECUTIVE_FAILURES,
+      onBreak,
+      onDegraded,
+      retries: RETRIES,
+    });
+    this.#fetchStatusPolicy = createRetryPolicy({
+      circuitBreakDuration,
+      degradedThreshold,
+      maximumConsecutiveFailures: MAX_CONSECUTIVE_FAILURES,
+      onBreak,
+      onDegraded,
+      retries: RETRIES,
+    });
+  }
+
+  /**
+   * Submit a deletion request.
+   *
+   * We use Segment for this request. Segment calls this deletion request a "regulation", and
+   * returns a "regulation ID" to keep track of this request and get status updates for it.
+   *
+   * @param metaMetricsId - The ID associated with the analytics data that we will be deleting.
+   * @returns The regulation ID for the deletion request.
+   */
+  async createDataDeletionRegulationTask(
+    metaMetricsId: string,
+  ): Promise<RegulationId> {
+    const response = await this.#createDataDeletionTaskPolicy.execute(() =>
+      this.#fetchWithTimeout(
+        `${this.#analyticsDataDeletionEndpoint}/regulations/sources/${
+          this.#analyticsDataDeletionSourceId
+        }`,
+        {
+          method: 'POST',
+          headers: { 'Content-Type': 'application/vnd.segment.v1+json' },
+          body: JSON.stringify({
+            regulationType: 'DELETE_ONLY',
+            subjectType: 'USER_ID',
+            subjectIds: [metaMetricsId],
+          }),
+        },
+      ),
+    );
+    return response.json();
+  }
+
+  /**
+   * Fetch the status of the given deletion request.
+   *
+   * @param deleteRegulationId - The Segment "regulation ID" for the deletion request to check.
+   * @returns The status of the given deletion request.
+   */
+  async fetchDeletionRegulationStatus(
+    deleteRegulationId: string,
+  ): Promise<CurrentRegulationStatus> {
+    const response = await this.#fetchStatusPolicy.execute(() =>
+      this.#fetchWithTimeout(
+        `${
+          this.#analyticsDataDeletionEndpoint
+        }/regulations/${deleteRegulationId}`,
+        {
+          method: 'GET',
+          headers: { 'Content-Type': 'application/vnd.segment.v1+json' },
+        },
+      ),
+    );
+    return response.json();
```

Comment:
It looks like the only property we use from this response is `data.regulation.overallStatus. Perhaps we should return that directly here, rather than passing along the entire response object that we have no use for.

That would also mean validating that the `overallStatus` property exists on the response. Right now we're not doing any validation, so this would fail if the response was shaped differently or was missing that value.